swift/Arrow/Sources/Arrow/ArrowCExporter.swift

```diff
@@ -61,12 +61,18 @@ public class ArrowCExporter {
             // deallocated
             self.arrowData = arrowData
             for arrowBuffer in arrowData.buffers {
-                data.append(arrowBuffer.rawPointer)
+                self.data.append(arrowBuffer.rawPointer)
             }
 
-            self.buffers = UnsafeMutablePointer(mutating: data)
+            self.buffers = UnsafeMutablePointer<UnsafeRawPointer?>.allocate(capacity: data.count)
+            self.buffers.initialize(from: &data, count: data.count)
```

Comment:
Should we use `self.data` here too?

```suggestion
            self.buffers = UnsafeMutablePointer<UnsafeRawPointer?>.allocate(capacity: self.data.count)
            self.buffers.initialize(from: &self.data, count: self.data.count)
```